lib/sbom/cyclonedx.ex

```diff
@@ -470,4 +465,41 @@ defmodule SBoM.CycloneDX do
     # credo:disable-for-next-line Credo.Check.Warning.UnsafeToAtom
     defp bom_struct(module, unquote(schema_version), attrs), do: struct(Module.concat([unquote(prefix), module]), attrs)
   end
+
+  @spec encode_json(map(), boolean()) :: binary()
+  defp encode_json(encodable, pretty)
+  # Jason selected
+  defp encode_json(encodable, pretty) when @json_module == Jason do
```

Comment:
Conditionally compile the function instead of adding a when clause.

This will also make the errors in the CI tests go away.